p11-kit/list-objects.c

```diff
@@ -202,8 +266,45 @@ print_object (p11_list_printer *printer,
 	};
 	CK_ULONG n_attrs = sizeof (attrs) / sizeof (attrs[0]);
 
+	allocated = p11_array_new (free);
+	if (allocated == NULL) {
+		p11_message (_("failed to allocate memory"));
+		return;
+	}
+
+	p11_kit_iter_get_attributes (iter, attrs, n_attrs);
+	for (i = 0; i < n_attrs; ++i) {
+		if (attrs[i].pValue != NULL_PTR ||
+		    attrs[i].ulValueLen == CK_UNAVAILABLE_INFORMATION)
+			continue;
+
+		if (attrs[i].ulValueLen >= SIZE_MAX) {
```

Comment:
`p11_buffer` has the overflow check. What about using it instead of doing that manually?

Reply:
Because the intent is much clearer and the code is simpler then using the abstraction p11_buffer. Also the example you provided would not work because after calling `p11_buffer_init (&buf, len);` the `buffer.len` would still be `0` and therefore you would first need to fill in the buffers and then call `p11_buffer_add (&buf, "\0", 1);` otherwise you are just adding '\0' at `buf.data[0]`. I dont see the point in using `p11_buffer` in this case.

Reply:
Ah, that's correct; it would be nice if the `p11_buffer` provides an API to reserve some more room, without appending data.